scipio/src/dma_file.rs

```diff
@@ -484,19 +484,25 @@ impl DmaFile {
 }
 
 #[cfg(test)]
-fn make_test_directory(test_name: &str) -> PathBuf {
+enum TestDirectoryKind {
+    TempFs,
+    StorageMedia,
+}
+
+#[cfg(test)]
+fn make_test_directories(test_name: &str) -> std::vec::Vec<(PathBuf, TestDirectoryKind)> {
+    let mut vec: std::vec::Vec<(PathBuf, TestDirectoryKind)> = Vec::new();
+
     // Scipio currently only supports NVMe-backed volumes formatted with XFS or EXT4.
-    // We therefore can't use the system-provided tmpfs for our test so instead we let the user
-    // decide what directory scipio should use to perform integration tests.
+    // We therefore let the user decide what directory scipio should use to host the unit tests in.
     // For more information regarding this limitation, see the README
-    match std::env::var("SCIPIO_TESTROOTDIR") {
+    match std::env::var("SCIPIO_TEST_POLLIO_ROOTDIR") {
         Err(_) => {
             eprintln!(
                 "Scipio currently only supports NVMe-backed volumes formatted with XFS \
-                or EXT4. To run io-related tests, please set SCIPIO_TESTROOTDIR to a NVMe-backed \
-                directory path in your environment."
+                or EXT4. To run io-related tests, please set SCIPIO_TEST_POLLIO_ROOTDIR to a \
```

Comment:
I think I am okay with that.

We can reconsider later. Is the message visible enough? `cargo test` usually doesn't show output unless you specify `--nocapture` My fear is that in practice nobody but us will run this, and gosh, even we can forget.

Reply:
This approach has the main advantage that every test run on both implementations "for free" and it'll be easy for new tests to do so as well.
The big caveat is as you said the risk that end up never running the poll mode tests which are really important. `cargo test` does indeed swallow the message which is a big issue in this case. 

Not sure what the right choice is.

Reply:
Ok. I spent some time thinking about this.

Let's try this way.

It would be nice if we could add least add this to a file people will read. This feels like a job for a getting started file, but for now I think the readme is enough. We also need to make sure that our CI always have this set.